src/libraries/System.Memory.Data/src/System/BinaryData.cs

```diff
@@ -230,9 +230,9 @@ public override unsafe string ToString()
         /// converted to.</typeparam>
         /// <param name="options">The <see cref="JsonSerializerOptions"/> to use when serializing to JSON.</param>
         /// <returns>The data converted to the specified type.</returns>
-        public T ToObjectFromJson<T>(JsonSerializerOptions? options = default)
+        public T? ToObjectFromJson<T>(JsonSerializerOptions? options = default)
```

Comment:
It's a source breaking change, but it's also accurately reflecting what's possible.  We've been taking such changes.

Reply:
We already have a GA version of this type that does not have the nullable annotation. Many Azure libraries have a dependency on this version. I just want to confirm that we are okay with the potential impact this would have on Azure SDK users.
/cc @KrzysztofCwalina @tg-msft @pakrym